### ISSUE TITLE
support ImageMinimizerWebpackPlugin

### ISSUE THREAD

**Issue by jerrykingxyz** [web-infra-dev/rspack#2843]
### What problem does this feature solve?

Supports all hooks used by ImageMinimizerWebpackPlugin.
https://github.com/webpack-contrib/image-minimizer-webpack-plugin

Here is the list of unsupported hooks
``` js
compilation.hooks.moduleAsset
compilation.hooks.assetPath
compilation.hooks.afterSeal
compilation.hooks.statsPrinter
stats.hooks.print
```

**Comment by hardfist** [comment]
we already supports https://www.npmjs.com/package/image-webpack-loader, so can  use it as workaround

**Comment by dartess** [comment]
`image-webpack-loader` uses `imagemin`

but `image-minimizer-webpack-plugin` can be used with `sharp` and `svgo`

surely this should be "closed this as completed"?

**Comment by OreQr** [comment]
Bump

**Comment by OreQr** [comment]
You can close it I managed to make it work with this: https://rspack.dev/guide/features/asset-module#using-optimizers-as-loaders

**Comment by dartess** [comment]
> it work with this

Yes, but loader will only process those images that were imported. The plugin also processed those images that were simply copied by the `HtmlWebpackPlugin` plugin.

Therefore, in `rspack`, it was necessary to additionally manually configure optimization via `transform` in `CopyRspackPlugin`, like this:

```js
transform: (input, absoluteFilename) => {
  if (absoluteFilename.includes('/affiliate/')) {
    return input;
  }
  switch (path.extname(absoluteFilename)) {
    case '.svg':
      return R.pipe(
        input.toString('utf-8'),
        content => svgo.optimize(content, svgoConfig),
        R.prop('data'),
      );
    case '.png':
      return sharp(input).png(sharpPngOptions).toBuffer();
    default:
      return input;
  }
},
```

So in an ideal world it would still be great to have support for `image-minimizer-webpack-plugin` as a plugin rather than as a loader.

**Comment by dartess** [comment]
bump

**Comment by dartess** [comment]
bump
